src/collaboration-platform/GitHubCollaborationPlatform.ts

```diff
@@ -17,25 +41,126 @@ export class GitHubCollaborationPlatform implements CollaborationPlatform {
   }
 
   async getPRForCommit(commit: Commit): Promise<PullRequestInfo | null> {
-    // TODO: Implement
-    return null;
+    const owner = (await repoPathToOwnerAndRepo(this.repoPath)).owner;
+    const repo = (await repoPathToOwnerAndRepo(this.repoPath)).repo;
+    try {
+      const {
+        data,
+      } = await this.octokit.repos.listPullRequestsAssociatedWithCommit({
+        owner,
+        repo,
+        commit_sha: commit.hash,
+      });
+      const prResult = data[0];
+      return {
+        number: prResult.number,
+        url: prResult.url,
+        title: prResult.title,
+        description: prResult.body,
+        isOutdated: prResult.head.sha !== commit.hash,
+      };
+    } catch (e) {
+      return null;
+    }
   }
 
   async getPR(prNumber: PullRequestID): Promise<PullRequestInfo | null> {
-    // TODO: Implement
-    return null;
+    const owner = (await repoPathToOwnerAndRepo(this.repoPath)).owner;
+    const repo = (await repoPathToOwnerAndRepo(this.repoPath)).repo;
+    try {
+      const prResult = this.octokit.pulls.get({
+        owner,
+        repo,
+        pull_number: prNumber,
+      });
+
+      return {
+        number: prNumber,
+        url: (await prResult).data.url,
+        title: (await prResult).data.title,
+        description: (await prResult).data.body,
```

Comment:
Code style: I think we should await on the first line instead of calling await repeatedly below. This follows the example in Octokit's docs:

```ts
const { data: pullRequest } = await octokit.pulls.get({
  owner: "octokit",
  repo: "rest.js",
  pull_number: 123,
});
```

```suggestion
      const { data: pullRequest } = await this.octokit.pulls.get({
        owner,
        repo,
        pull_number: prNumber,
      });

      return {
        number: prNumber,
        url: pullRequest.url,
        title: pullRequest.title,
        description: pullRequest.body,
```